### PR TITLE
mpv: fix cross compilation

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/mpv/default.nix [hunk 1]
```diff
@@ -24,6 +24,7 @@
 , waylandSupport ? stdenv.isLinux
   , wayland
   , wayland-protocols
+  , wayland-scanner
```
Comment (AndersonTorres) [inline pkgs/applications/video/mpv/default.nix:27]: How exactly it fixes cross compilation?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/mpv/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/video/mpv/default.nix:27]:
```
mpv-aarch64-unknown-linux-gnu> Checking for Wayland                                                      : wayland-scanner not found
mpv-aarch64-unknown-linux-gnu> You manually enabled the feature 'wayland', but the autodetection check failed.
```

Comment (rnhmjoj) [reply pkgs/applications/video/mpv/default.nix:27]: I had fixed this in #187797, which is still in staging. Not sure what will happen now.

Comment (SuperSandro2000) [reply pkgs/applications/video/mpv/default.nix:27]: It is still in staging-next and it will probably cause a merge conflict somewhere.

**Pull request merged**